nipype/algorithms/confounds.py

```diff
@@ -561,16 +602,25 @@ def _run_interface(self, runtime):
 
         components_file = os.path.join(os.getcwd(),
                                        self.inputs.components_file)
+        components_header = self._make_headers(components.shape[1])
         np.savetxt(
             components_file,
             components,
             fmt=b"%.10f",
             delimiter='\t',
-            header=self._make_headers(components.shape[1]),
+            header='\t'.join(components_header),
             comments='')
+        self._results['components_file'] = os.path.abspath(
+            self.inputs.components_file)
 
-        if self.inputs.pre_filter and self.inputs.save_pre_filter:
-            pre_filter_file = self._list_outputs()['pre_filter_file']
+        save_pre_filter = self.inputs.save_pre_filter
```

Comment:
I can't follow very well the block starting in line 623, wouldn't this be equivalent?
```suggestion
        save_pre_filter = False
        if self.inputs.pre_filter in ['polynomial', 'cosine']:
            save_pre_filter = self.inputs.save_pre_filter
```